### PR TITLE
Add support for multi-line messages. Close #20

### LINKED ISSUE #20: Add support for multi-line messages

### PR DESCRIPTION
**Pull request by artoale**
Shamelessly copied from the work on #11

### CODE REVIEW

**Comment by artoale** [comment]
@steelbrain pinging you in :)

File: lib/init.coffee [hunk 1]
```diff
@@ -46,20 +56,19 @@ module.exports =
             '--quiet',
             tmpFilePath
           ]
-          return helpers.exec(@executablePath, params, {cwd}).then (stdout) ->
-            regex = /Line\s(\d+), E:([^:]+):\s(.+)/
-            lines = stdout.split('\n').filter (line) ->
-              line.indexOf('Line') is 0
+          return helpers.exec(@executablePath, params, {cwd}).then (stdout) =>
+            regex = XRegExp '^(?<line>\\d+), E:(?<code>[^:]+): (?<message>.+)$',
```
Comment (steelbrain) [inline lib/init.coffee:60]: We have `atom-linter::parse` for this. It uses `XRegExp` too

**Comment by steelbrain** [comment]
The parse method doesn't support flags at the moment, let me fix that

**Comment by artoale** [comment]
@steelbrain note that I'm using `XRegExp::forEach` as well, which is pretty handy :)

File: lib/init.coffee [hunk 2]
```diff
@@ -46,20 +56,19 @@ module.exports =
...
-                text: text,
-                type: 'error',
-                filePath: filePath,
-                range: helpers.rangeFromLineNumber(editor, line - 1)
-              }
+
+            messages = []
+
+            lines.forEach (msg) =>
+              XRegExp.forEach msg, regex, (match, i) =>
```
Comment (steelbrain) [inline lib/init.coffee:70]: Do you ever get more than one error per line? If not, then you don't need this

Comment (artoale) [inline lib/init.coffee:70]: I assumed so, but need to doublecheck - good catch!

**Comment by artoale** [comment]
@steelbrain I'm trying to re-implement this using `helpers.parse`, but it has a couple of issue:
- I can't do arbitrary match (which means I can't match for error code, which will be useful for e.g. error suppression of specific error codes)
- `parse` is doing multi-line splitting, which makes it impossible (AFAICT) to have multi-line messages, which is the whole point of this PR :) 

Also, I've checked, and yes, gjslint does report multiple messages per line, but they're all separated so I don't strictly speaking "need" XRegExp.forEach, but it's useful to strip out non-matching lines without an "if" (and accounting for weird input I might have never stumbled upon)

Any ideas?

**Comment by artoale** [comment]
Also, ranges generated by `parse` don't include the whole line, but just the first character, which changes the current behavior

**Comment by steelbrain** [comment]
In that case you have a LGTM after you confirm this works

**Comment by artoale** [comment]
Yeah, I've been using this on a local branch for a while now, with no issue AFAICT.

**Pull request merged**